cleo_plugins/Text/Text.cpp

```diff
@@ -439,10 +448,11 @@ class Text
 		auto posY = OPCODE_READ_PARAM_FLOAT();
 		OPCODE_READ_PARAM_STRING_FORMATTED(text);
 
-		// new GXT label
+		// new generic GXT label
 		// includes unprintable character, to ensure there will be no collision with user GXT labels
-		char gxt[8] = { 0x01, 'C', 'L', 'E', 'O', '_', 0x01, 0x00 };
-		gxt[6] += CTheScripts::NumberOfIntroTextLinesThisFrame; // unique label for each possible entry
+		char gxt[9] = { 0x01, 'C', 'L', 'E', 0x00, 0x00, 0x00, 0x00, 0x00 }; // enough space for even worst case scenario
+		_itoa(genericLabelCounter, gxt + 4, 36); // 0xFFFF -> "1ekf"
+		genericLabelCounter++;
 
 		textManager.AddFxt(gxt, text);
```

Comment:
are these keys ever removed from textManager, or we keep adding them?

Reply:
These are kept, but names are procedural. So total sum of keys never exceeds max count of display_text_formatted calls in single draw frame.

Reply:
ok, and why we convert the counter to text if previously we added it to the last byte?

in other words, why not:

```
gxt[6] += genericLabelCounter++;
```

Reply:
genericLabelCounter is cleared once peer render frame, so in theory multiple scripts can print total sum of more than 255 texts.

Reply:
does the game support than many draws? I thought the limit is 96

https://github.com/sannybuilder/docs/blob/en/scm-documentation/gta-limits.md

Reply:
eh? 96 * 96 = 9216 draws?

Reply:
![image](https://github.com/user-attachments/assets/2f7104a7-969e-4618-8efb-9eaaa66f6125)
32 scripts drawing 96 texts each. 
Works, but as usually in case of GTA it is hard to tell if some array is not getting overflown.
Anyway, I'm not the one clearing drawn texts counter. Not sure if they copy text entries to some other array or actually just draw accumulated requests after processing of each script.